src/CashuWallet.ts

```diff
@@ -254,18 +255,7 @@ class CashuWallet {
 	 * @param options.requireDleq? will check each proof for DLEQ proofs. Reject the token if any one of them can't be verified.
 	 * @returns New token with newly created proofs, token entries that had errors
 	 */
-	async receive(
-		token: string | Token,
-		options?: {
-			keysetId?: string;
-			outputAmounts?: OutputAmounts;
-			proofsWeHave?: Array<Proof>;
-			counter?: number;
-			pubkey?: string;
-			privkey?: string;
-			requireDleq?: boolean;
-		}
-	): Promise<Array<Proof>> {
+	async receive(token: string | Token, options?: SendOptions): Promise<Array<Proof>> {
```

Comment:
I don't think that we should have one big merged type for the options of different methods. It does avoid repeating code in types, however it is now no longer clear what options are available on a method. For example the type here suggests that `options.includeFees` would be a valid options for the `receive` method, but it is actually not.

Reply:
so we should have an object destructuring that pull only the field that function need,

Reply:
No, the problem is that the type signature of `receive` now looks like `receive(token: string | Token, options?: SendOptions): Promise<Array<Proof>>`. This makes someone believe that this function would accept something like `options.includeFees` because `includeFees` is on type `SendOptions`. However this function does not accept that option.

The type for `options` on `receive` should not include any option that is not supported. If you want to solve this with a single type, please use `Pick`, but TBH I believe having separate option types would be better.

Reply:
so each function should have it's own separate type options.

Reply:
yes i think it's best to have separate types for each function